hydradx/tests/test_stableswap.py

```diff
@@ -629,3 +629,55 @@ def test_buy_sell_spot(
         raise AssertionError(f'Sell spot R1 ({r2_per_r1}) != execution price ({ex_price_r1}).')
     if r1_per_r2 != pytest.approx(ex_price_r2):
         raise AssertionError(f'Buy spot R2 ({r1_per_r2}) != execution price ({ex_price_r2}).')
+
+
+@given(
+    st.lists(asset_quantity_strategy, min_size=2, max_size=2),
+    st.floats(min_value=0.0001, max_value=0.50),
+    st.integers(min_value=10, max_value=100000)
+)
+def test_share_prices(assets, fee, amp):
+    tokens = {'USDT': mpf(assets[0]), 'USDC': mpf(assets[1])}
+    pool = StableSwapPoolState(tokens, mpf(amp), trade_fee=mpf(fee))
+    spot = pool.share_price('USDT')
+    add_liq_price = pool.add_liquidity_spot('USDT')
+    if add_liq_price <= spot:
+        raise AssertionError('Add liquidity price should be higher than spot price.')
+    buy_shares_price = pool.buy_shares_spot('USDT')
+    if buy_shares_price <= spot:
+        raise AssertionError('Buy shares price should be higher than spot price.')
+    remove_liq_price = pool.remove_liquidity_spot('USDT')
+    if remove_liq_price >= spot:
+        raise AssertionError('Remove liquidity price should be lower than spot price.')
+    withdraw_asset_price = pool.withdraw_asset_spot('USDT')
+    if withdraw_asset_price >= spot:
+        raise AssertionError('Withdraw asset price should be lower than spot price.')
+
+
+@given(
+    st.lists(asset_quantity_strategy, min_size=2, max_size=2),
+    st.floats(min_value=0.0001, max_value=0.10),
+    st.integers(min_value=10, max_value=100000)
+)
+def test_share_prices_optional(assets, fee, amp):
```

Comment:
Most of this test is just duplicating the previous test, but with less range. I think only the last two clauses are needed here.

Reply:
I will remove test_share_prices_optional, it was mainly observing what the relationship was between buy_shares_price and add_liq_price, and same for removing liquidity. Those are not actual requirements for correctness.